messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java

```diff
@@ -161,8 +162,41 @@ public ActiveMQServerService(Configuration configuration, PathConfig pathConfig)
         return dataSource;
     }
 
-    Map<String, JChannel> getChannels() {
-        return channels;
+    JChannel getChannel(String name) throws Exception {
+        if (channels.containsKey(name)) {
+            return channels.get(name);
+        }
+        if (jgroupsChannels.containsValue(name)) {
+            Optional<String> groupName = jgroupsChannels.entrySet().stream().filter(entry -> name.equals(entry.getValue())).map(entry -> entry.getKey()).findFirst();
+            if (groupName.isPresent()) {
+                final String group;
+                if(groupName.get().startsWith("discovery") || groupName.get().startsWith("broadcast")) {
+                    group = groupName.get().substring(9);
+                } else {
+                    group = groupName.get();
+                }
+                if (configuration.getDiscoveryGroupConfigurations().containsKey(group)) {
+                    try {
+                        configuration.getDiscoveryGroupConfigurations().get(group).getBroadcastEndpointFactory().createBroadcastEndpoint();
```

Comment:
The purpose of JChannelManager/JChannelWrapper is to ensure that only one user of the channel ever connects, and that the channel is only closed when the last user is done with it.  JGroupsBroadcastEndpointFactory.createBroadcastEndpoint() results in the reference being tracked by JChannelManager.  Since this BroadcastEndpoint is never closed, the reference count will never decrement, thus this will throw off the (admittedly brittle) reference counting, and the channel will never be closed.